File: 소프티어/JavaScript/Lv3/성적평균.js
const readline = require('readline')
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
})

let input = [];
rl.on('line', function(line){
    input.push(line.split(' ').map((i, _) => Number(i)))
}).on('close', function() {
    const [N, K] = input[0];
    const score = input[1];
    const arr = input.slice(2);

    const sum_arr = [0];

    for (let i = 0; i < N; i++){
        sum_arr.push(sum_arr[i] + score[i])
    }

    for (let i = 0; i < K; i++){
        const [s, e] = arr[i];
        let total = sum_arr[e] - sum_arr[s - 1]
        total /= e - s + 1
        const tmp = total.toFixed(2);

        console.log(tmp)
    }
})

File: 소프티어/Python/Lv3/성적평균.py
import sys
input = sys.stdin.readline

# N명의 학생들의 성적이 학번순서대로 주어짐
# 학번 구간 [A, B]가 주어졌을때 학생들 성적의 평균을 구해야함

# 학생수 N, 구간 수 K
N, K = map(int, input().split())

results = [0] + list(map(int, input().split()))

for i in range(K):
    A, B = map(int, input().split())
    total = 0
    people = B - A + 1
    for j in range(A, B + 1):
        total += results[j]

    print("{:.2f}".format(total / people))
